packer/provisioner.go

```diff
@@ -177,29 +177,35 @@ func (p *PausedProvisioner) pausingNoUpdate(ctx context.Context) error {
 
 func (p *PausedProvisioner) pausingWithUpdates(ctx context.Context, ui packersdk.Ui) error {
 	updateTime := 10
-	timeTicker := time.NewTicker(time.Duration(updateTime) * time.Second)
-	TotalTime := p.PauseBefore.Seconds()
+	ticker := time.NewTicker(time.Duration(updateTime) * time.Second)
+	defer ticker.Stop()
+	totalTime := p.PauseBefore.Seconds()
+
+	ctx, cancel := context.WithTimeout(context.Background(), p.PauseBefore)
```

Comment:
Here, we should use the incoming context; so that if it is cancelled (ex: ctrl + c on unix), this returns immediately

```suggestion
	ctx, cancel := context.WithTimeout(ctx, p.PauseBefore)
```